packages/mobile/src/components/share-drawer/useShareToStory.tsx

```diff
@@ -435,29 +442,68 @@ export const useShareToStory = ({
         const videoUri = `file://${storyVideoPath}`
         try {
           if (platform === 'instagram') {
-            await pasteToInstagramApp(videoUri, stickerUri)
+            // On Android, don't await Instagram share to prevent app from coming back to foreground
+            // react-native-share resolves too quickly before Instagram fully opens
+            if (Platform.OS === 'android') {
+              pasteToInstagramApp(videoUri, stickerUri).catch((error) => {
+                // Only handle errors, don't block on success
+                handleError(platform, error, 'Error at share to app step')
+              })
+              // Give Instagram time to open before cleanup
```

Comment:
do we not got any callback we can use? or do this when the user returns to audius instead of on a delay after open?